lahja/asyncio/endpoint.py

```diff
@@ -528,3 +527,12 @@ def subscribe(self,
                     # mypy cannot reconcile BaseEvent with TStreamEvent
                     self._queues[event_type].remove(queue)  # type: ignore
                     break
+
+
+class Endpoint(AsyncioEndpoint):
+    def __init__(self) -> None:
+        warnings.warn(
+            "`lahja.Endpoint` has been deprecated in favor of `lahja.AsyncioEndpoint`",
+            category=DeprecationWarning,
+        )
+        super().__init__()
```

Comment:
At this stage of the library, with us most likely being the only users, I think we don't need to issue deprecation warnings and just break things as we see the need for it.